Makefile

```diff
@@ -0,0 +1,68 @@
+VERSION := 0.16.5
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect sources
+ifneq ($(AMALGAMATED),1)
+	SRC := $(wildcard lib/src/*.c)
+	# do not double-include amalgamation
+	SRC := $(filter-out lib/src/lib.c,$(SRC))
+else
+	# use amalgamated build
+	SRC := lib/src/lib.c
+endif
+OBJ := $(SRC:.c=.o)
+
+# define default flags, and override to append mandatory flags
+CFLAGS ?= -O3
```

Comment:
Just curious, why did you choose `-O3`? Did you do any tests against `-O2`?

Reply:
I faithfully copied this from https://github.com/tree-sitter/tree-sitter/blob/1635aab801152b8b36975dbc6cd02ea01ddbf778/script/build-lib#L10-L12

This is an existing project policy for default cflags, and I'm focused here on only creating a Makefile build with parity to the existing script.